safe-room.lic

```diff
@@ -26,9 +26,22 @@ class SafeRoom
 
     return unless args.force || need_healing?
 
-    if use_pc_empath?(settings.safe_room_id, settings.safe_room_empath)
-      ensure_copper_on_hand(settings.safe_room_tip_threshold || 0, settings)
-      tip(settings.safe_room_tip_threshold, settings.safe_room_tip_amount, settings.safe_room_empath)
+    pc_empaths = settings.safe_room_empaths || []
+    if settings.safe_room_id && settings.safe_room_empath
+      # Add to pc_empaths list if its not in there already
+      pc_empaths << { 'name' => settings.safe_room_empath, 'id' => settings.safe_room_id } unless pc_empaths.any? { |empath| empath['name'] == settings.safe_room_empath && empath['id'] == settings.safe_room_id }
+    end
+
+    if !pc_empaths.empty?
+      # Sort pc empaths list by nearest first
+      sort_destinations(pc_empaths.map { |empath| empath['id'] })
+        .map { |room| pc_empaths.find { |empath| empath['id'] == room } }
+        .each do |empath|
+        next unless use_pc_empath?(empath['id'], empath['name'])
```

Comment:
Could this be extracted into a `reject` clause before the `each`?

Reply:
I dont think so - its re ordering an array of hashes based on the array of room numbers returned from sort destinations, but not discarding anything.

Reply:
I could look into replacing `find` with `select` or `reject` to cover the edge case where two empaths share the same room number. Let me know if I misunderstood your question.

Reply:
Rereading the code, I think I'm wrong.

I hadn't thought about the case where two empaths share the same room. Since we're sorting destinations based on distance, I assume the two empaths would always be adjacent in the ordering, so we shouldn't need to handle that in a special way.